org.ultimatepp.TheIDE.yml

```diff
@@ -0,0 +1,114 @@
+id: org.ultimatepp.TheIDE
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions: ["org.freedesktop.Sdk.Extension.llvm17"]
+command: run-ide.sh
+finish-args:
+  - --socket=x11
+  - --share=network
+  - --share=ipc
+  - --filesystem=host
+  - --talk-name=org.freedesktop.Flatpak
+modules:
+  - name: libnotify
+    buildsystem: meson
+    config-opts:
+      - -Dtests=false
+      - -Dintrospection=disabled
+      - -Dman=false
+      - -Dgtk_doc=false
+      - -Ddocbook_docs=disabled
+    sources:
+      - type: archive
+        url: https://download.gnome.org/sources/libnotify/0.8/libnotify-0.8.3.tar.xz
+        sha256: ee8f3ef946156ad3406fdf45feedbdcd932dbd211ab4f16f75eba4f36fb2f6c0
+
+  - name: upp
+    buildsystem: simple
+    build-commands: 
+      - printf "#define flagFlatpak 1\\n" > uppsrc/uppconfig.h
+      - make -f Makefile -j $FLATPAK_BUILDER_N_JOBS
+      - printf "\\n" > uppsrc/uppconfig.h
+      - make -f umkMakefile.in -j $FLATPAK_BUILDER_N_JOBS
+      - mv ide theide
+      - install -Dm755 theide /app/bin/theide
+      - install -Dm755 umk /app/bin/umk
+      - install -D uppsrc/ide/icon.svg /app/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+      - install -D uppsrc/ide/theide.desktop /app/share/applications/${FLATPAK_ID}.desktop
+      - desktop-file-edit --set-key=Icon --set-value=${FLATPAK_ID} /app/share/applications/${FLATPAK_ID}.desktop
+      - mkdir -p /app/src
+      - mkdir -p /app/lib
+      - mkdir -p /app/bm
+      - printf "\\n" > uppsrc/uppconfig.h
+      - rm -f /app/src/ver.txt && touch /app/src/ver.txt
+      - printf "2023.2" > /app/src/ver.txt
+      - cp -r examples /app/src
+      - cp -r reference /app/src
+      - cp -r tutorial /app/src
+      - cp -r uppsrc /app/src
+      - cp .clang-format /app/src/uppsrc/.clang-format
+      - cp uppbox/Scripts/GCCMK.bm /app/bm/GCC.bm
+      - cp run-ide-install-host-deps.sh /app/src/run-ide-install-host-deps.sh
+      - cp -r /usr/lib/sdk/llvm17/lib /app
+    sources:
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/archive/refs/tags/2023.2.tar.gz
+        sha256: 8e0f02d2cdcd23f7774e9c576a8cc7afa50b97232046d09940e386cb89dcf286
+      - type: file
+        path: run-ide-install-host-deps.sh
+      - type: patch
+        path: patches/icon.patch
+      - type: patch
+        path: patches/theide.patch
+  
+  - name: upp-term
+    buildsystem: simple
+    build-commands:
+      - mv 3p/Terminal/PtyProcess app/PtyProcess
+      - mv 3p/Terminal/PtyAgent app/PtyAgent
+      - mv 3p/Terminal/Terminal app/Terminal
+      - /app/bin/umk app/,3p/uppsrc UppTerm /app/bm/GCC.bm -brv +GUI,SHARED upp-term
+      - install -Dm755 upp-term /app/bin/upp-term
+      - rm -f /app/bin/umk
+      - rm -rf /app/bm/
```

Comment:
It looks like when I put those into cleanup step, it doesn't work. After compiling UppTerm we do not longer need those files anymore in flatpak.